Tasks/ServiceFabricDeploy/deploy.ps1

```diff
@@ -0,0 +1,87 @@
+# For more information on the VSTS Task SDK:
+# https://github.com/Microsoft/vsts-task-lib
+
+[CmdletBinding()]
+param()
+
+Trace-VstsEnteringInvocation $MyInvocation
+try {
+    # Import the localized strings. 
+    Import-VstsLocStrings "$PSScriptRoot\task.json"
+    
+    # Load utility functions
+    . "$PSScriptRoot\utilities.ps1"
+    
+    # Collect input values
+    $publishProfilePathSearchPattern = Get-VstsInput -Name publishProfilePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForPublishProfile -ArgumentList $publishProfilePathSearchPattern) 
+    $publishProfilePath = Find-VstsFiles -LegacyPattern $publishProfilePathSearchPattern
+    Assert-SingleItem $publishProfilePath $publishProfilePathSearchPattern
+    Assert-VstsPath -LiteralPath $publishProfilePath -PathType Leaf
+    Write-Host (Get-VstsLocString -Key FoundPublishProfile -ArgumentList $publishProfilePath)
+    
+    $applicationPackagePathSearchPattern = Get-VstsInput -Name applicationPackagePath -Require
+    Write-Host (Get-VstsLocString -Key SearchingForApplicationPackage -ArgumentList $applicationPackagePathSearchPattern)
+
+    # Workaround for bug in Find-VstsFiles that doesn't handle the IncludeDirectories parameter correctly. Modifies the search pattern to look for app manifest file instead of the directory
+    # and then get the directory of that file.
+    $applicationPackagePathSearchPattern = [System.IO.Path]::Combine($applicationPackagePathSearchPattern, "ApplicationManifest.xml")
+    $applicationPackagePath = Find-VstsFiles -LegacyPattern $applicationPackagePathSearchPattern
+    Assert-SingleItem $applicationPackagePath $applicationPackagePathSearchPattern
+    $applicationPackagePath = [System.IO.Path]::GetDirectoryName($applicationPackagePath)
+    Assert-VstsPath -LiteralPath $applicationPackagePath -PathType Container
+    Write-Host (Get-VstsLocString -Key FoundApplicationPackage -ArgumentList $applicationPackagePath)
+
+    $aadUserName = Get-VstsInput -Name aadUserName -Require
+    Write-Host "userName: $aadUserName"
+    
+    $aadUserPassword = Get-VstsInput -Name aadUserPassword -Require
+    # Do NOT write value of password to the output
+
+    $publishProfile = Read-PublishProfile $publishProfilePath
+    $clusterConnectionParameters = $publishProfile.ClusterConnectionParameters
+    
+    if (-not $clusterConnectionParameters.ContainsKey("AzureActiveDirectory") -or $clusterConnectionParameters.Item("AzureActiveDirectory") -eq $false)
```

Comment:
In general, you might want to double check that the syntax you use for hashtables and xml works on PowerShell 3. All of our tasks support as far back as that version. There has been discussion of adding a minimum PowerShell version to the handler section of the task.json, but is not there today. Otherwise if there is a reason it can't be supported back to PowerShell 3, then the task should fail gracefully (e.g. check psversiontable and error out with a good message).

I can't remember if the syntax for accessing hashtable values using property syntax (i.e. `$hash.MyKey` over `$hash['MyKey']`) is supported in PS3. If not, then one option to enable the consumer to still use the property syntax is to use `New-Object -TypeName psobject -Property $hash` which will return an object will properties.